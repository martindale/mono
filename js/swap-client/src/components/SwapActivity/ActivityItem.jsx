import React, { useCallback } from 'react'
import { Button, Grid, Stack } from '@mui/material'
import { getStringFromDate, SWAP_STATUS } from '../../utils/helpers'
import styles from '../../styles/ActivityItem.module.css'
import RotateLeftIcon from '@mui/icons-material/RotateLeft'
import { activitiesStore } from '../../syncstore/activitiesstore'
import { walletStore } from '../../syncstore/walletstore'

export const ActivityItem = ({ activity, index, handleClick }) => {
  const link = 'https://ordinals.com/tx/' + activity.hash

  const onCancelSwap = useCallback((e) => {
    e.stopPropagation()
    activitiesStore.dispatch({ type: 'CANCEL_SWAP', payload: index })
    if (activity.baseAsset === 'BTC') walletStore.dispatch({ type: 'ADD_NODE_BALANCE', payload: activity.baseQuantity })
    else if (activity.baseAsset === 'ETH') walletStore.dispatch({ type: 'ADD_WALLET_BALANCE', payload: activity.baseQuantity })
    else walletStore.dispatch({ type: 'ADD_NFT_BALANCE', payload: { type: activity.baseAsset, balance: 1 } })
  }, [activity, walletStore, activitiesStore])

  return (
    <Grid container direction='row' style={{ marginTop: 0 }} className={`${styles['activity-item']} activity-item`} onClick={handleClick}>
      <Grid container item xs={7}>
        <Stack direction='row' spacing={1}>
          <RotateLeftIcon />
          <Stack direction='column' textAlign='left'>
            <span className='activity_status'>{SWAP_STATUS[activity.status]}</span>
            <span className='grey'>{getStringFromDate(activity.createdDate)}</span>
            {activity.status === 5 && activity.hash.slice(0, 20) + '...'}
          </Stack>
        </Stack>
      </Grid>
      <Grid item xs={5}>
        <Stack direction='column' textAlign='right'>
          <span>- {activity.baseQuantity.toFixed(5).replace(/[.,]0+$/, '')} <span className='grey'>{activity.baseAsset.toLowerCase()}</span></span>
          <span>+ {activity.quoteQuantity.toFixed(5).replace(/[.,]0+$/, '')} <span className='grey'>{activity.quoteAsset.toLowerCase()}</span></span>
          {activity.status < 5 && <span><Button onClick={onCancelSwap} className={styles['cancel-btn']}>
            Cancel
                                        </Button>
                                  </span>}
        </Stack>
      </Grid>
    </Grid>
  )
}
