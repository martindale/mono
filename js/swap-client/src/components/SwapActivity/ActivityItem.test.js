import { ActivityItem } from './ActivityItem'
import { create, act } from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'

describe('AccessOptionComponent', () => {
  const initialState = {
  }
  const mockStore = configureStore()
  let store
  it('should work', () => {
    let tree
    const activity = {
      hash: '',
      baseAsset: 'BTC',
      baseQuantity: 0,
      createdDate: {
        year: 2000,
        month: 1,
        day: 1
      },
      status: 0,
      quoteAsset: 'ETH',
      quoteQuantity: 0
    }
    const handleClick = jest.fn()

    act(() => {
      store = mockStore(initialState)
      tree = create(
        <Provider store={store}>
          <ActivityItem
            activity={activity}
            index={0}
            handleClick={handleClick}
          />
        </Provider>
      )
    })

    expect(tree).toMatchSnapshot()
  })
  afterAll(() => jest.resetModules())
})
