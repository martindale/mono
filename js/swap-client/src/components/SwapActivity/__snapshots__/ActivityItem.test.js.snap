// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccessOptionComponent should work 1`] = `
<div
  className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
  onClick={[MockFunction]}
  style={
    {
      "marginTop": 0,
    }
  }
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-8 css-1dxyl14-MuiGrid-root"
  >
    <div
      className="MuiStack-root css-w4z10b-MuiStack-root"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="RotateLeftIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7.11 8.53 5.7 7.11C4.8 8.27 4.24 9.61 4.07 11h2.02c.14-.87.49-1.72 1.02-2.47zM6.09 13H4.07c.17 1.39.72 2.73 1.62 3.89l1.41-1.42c-.52-.75-.87-1.59-1.01-2.47zm1.01 5.32c1.16.9 2.51 1.44 3.9 1.61V17.9c-.87-.15-1.71-.49-2.46-1.03L7.1 18.32zM13 4.07V1L8.45 5.55 13 10V6.09c2.84.48 5 2.94 5 5.91s-2.16 5.43-5 5.91v2.02c3.95-.49 7-3.85 7-7.93s-3.05-7.44-7-7.93z"
        />
      </svg>
      <div
        className="MuiStack-root css-p6dskh-MuiStack-root"
      >
        <span>
          Submitting order
        </span>
        <span
          className="grey"
        >
          Feb 1, 0
        </span>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
  >
    <div
      className="MuiStack-root css-9yuikb-MuiStack-root"
    >
      <span>
        - 
        0.00
         
        <span
          className="grey"
        >
          btc
        </span>
      </span>
      <span>
        + 
        0.00
         
        <span
          className="grey"
        >
          eth
        </span>
      </span>
      <span>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          Cancel
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </span>
    </div>
  </div>
</div>
`;
